api/v1alpha1/stardoguser_types.go

```diff
@@ -18,18 +18,26 @@ package v1alpha1
 
 import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	types "k8s.io/api/core/v1"
 )
 
 // EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
 // NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
 
 // StardogUserSpec defines the desired state of StardogUser
 type StardogUserSpec struct {
-	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
-	// Important: Run "make" to regenerate code after modifying this file
+	// User describes a Stardog username
+	User string `json:"user,omitempty"`
+	// StardogUserPasswordSpec describes the password of a Stardog user
+	Password StardogUserPasswordSpec `json:"password,omitempty"`
```

Comment:
This seems an unnecessary nesting. Why not directly reference the secret?
```suggestion
	CredentialSecretRef types.SecretKeySelector `json:"credentialSecretRef,omitempty"`
```

Reply:
we could have a secret that would incorporate multiple user credentials... I think it's not a bad idea to have a bit of flexibility.

Reply:
I don't get your point...

Reply:
I wanted to delimit the password from the secret... but we could flatten it yes

Reply:
Well, the YAML with your initial proposal would look like this:
```yaml
spec:
  password:
    secretKeyRef:
      name: my-secret
      key: password
```
With my proposal, it looks like this:
```yaml
spec:
  credentialSecretRef:
    name: my-secret
    key: password
```
In neither approaches is it possible to specify multiple secrets. But in both approaches it's possible to have a single secret with multiple user passwords/tokens, provided they are distinguished by different `key`s.

Reply:
Alternatively, we could store both Username and Password in the secret. Since we already have a `spec.username` property, the `metadata.name` property is not the one that is used to authenticate against stardog. In that case I would change the type of `credentialSecretRef` to a plain string, and the operator gets it from the referenced secret with pre-defined keys, very similar to how secrets of type `kubernetes.io/tls` with Ingresses work.